src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -146,90 +191,142 @@ private async ValueTask LoadOrReloadProjectsAsync(ImmutableSegmentedList<Project
         }
     }
 
+    /// <summary>
+    /// Loads a project in the MSBuild host.
+    /// </summary>
+    protected abstract Task<(RemoteProjectFile projectFile, bool hasAllInformation, BuildHostProcessKind preferred, BuildHostProcessKind actual)?> TryLoadRemoteProjectAsync(
```

Comment:
nit: rename to TryLoadProjectInMSBuildHostAsync.  Then you don't need the doc comment.